napari/_qt/qt_viewer.py

```diff
@@ -341,6 +366,7 @@ def _add_layer(self, layer):
         vispy_layer.node.parent = self.view.scene
         vispy_layer.order = len(self.viewer.layers) - 1
         self.layer_to_visual[layer] = vispy_layer
+        self._on_active_layer_change(None)
```

Comment:
Is this call really need? The change of the active layer should be triggered by already connected signals. Maybe it is needed only when adding the first layer?

Reply:
Yes, this is the `_add_layer` method and it's needed when adding the layer

Reply:
But add layers change the active layer (deselect the previous one), so why next call of this function is need?

Reply:
maybe the previous one is already not selected

Reply:
Then cached value should connect `None`